Need type annotation for predefined dict generic (3.9)

<!--
  If you're new to mypy and you're not sure whether what you're experiencing is a mypy bug, please see the "Question and Help" form
  instead.
-->

**Bug Report**
mypy errors on defining a variable to a python 3.9 dict whose type signature has been previously stored in a variable

<!--
Note: If the problem you are reporting is about a specific library function, then the typeshed tracker is better suited
for this report: https://github.com/python/typeshed/issues
-->

**To Reproduce**
Using `mypy bug.py --strict`
Error stems from disallow untyped generics, I think
```py
Block = dict[str, str]

def main() -> Block:

    out = Block()  # error: Need type annotation for "out" (hint: "out: Dict[<type>, <type>] = ...")

    return out

if __name__ == "__main__":
    main()
```

**Expected Behavior**

Mypy should be able to read the type information stored in `Block` and read it as that type being init, as it can read the type information for a return signature

<!--
  How did you expect your project to behave?
  It’s fine if you’re not sure your understanding is correct.
  Write down what you thought would happen. If you just expected no errors, you can delete this section.
-->

**Actual Behavior**
mypy raises a untyped generic error that out needs a type annotation:
`error: Need type annotation for "out" (hint: "out: Dict[<type>, <type>] = ...")`

<!--
  Did something go wrong?
  Is something broken, or not behaving as you expected?
-->


**Your Environment**

<!-- Include as many relevant details about the environment you experienced the bug in -->

- Mypy version used: 0.910
- Mypy command-line flags: --strict
- Mypy configuration options from `mypy.ini` (and other config files): none present.
- Python version used: 3.9.6
- Operating system and version: Linux 5.10

<!--
You can freely edit this text, please remove all the lines
you believe are unnecessary.
-->

I'm starting to take a look at this issue and I made an example of my own. Interestingly, there doesn't seem a problem for `list`.

- ex11093/test3.py

```py
from typing import Dict, List

D1 = dict[str, str]
D2 = Dict[str, str]
x1 = D1()
x2 = D2()
x3 = dict[str, str]()

L1 = list[str]
L2 = List[str]
y1 = L1()
y2 = L2()
y3 = list[str]()
```

- command

```sh
$ mypy --strict --show-error-codes ex11093/test3.py 
ex11093/test3.py:5: error: Need type annotation for "x1" (hint: "x1: Dict[<type>, <type>] = ...")  [var-annotated]
ex11093/test3.py:6: error: Need type annotation for "x2" (hint: "x2: Dict[<type>, <type>] = ...")  [var-annotated]
Found 2 errors in 1 file (checked 1 source file)
```

--- a/mypy/semanal.py
+++ b/mypy/semanal.py
@@ -5842,7 +5842,11 @@ class SemanticAnalyzer(
             expr.analyzed = PromoteExpr(target)
             expr.analyzed.line = expr.line
             expr.analyzed.accept(self)
-        elif refers_to_fullname(expr.callee, "builtins.dict"):
+        elif refers_to_fullname(expr.callee, "builtins.dict") and not (
+            isinstance(expr.callee, RefExpr)
+            and isinstance(expr.callee.node, TypeAlias)
+            and not expr.callee.node.no_args
+        ):
             expr.analyzed = self.translate_dict_call(expr)
         elif refers_to_fullname(expr.callee, "builtins.divmod"):
             if not self.check_fixed_args(expr, 2, "divmod"):

--- a/test-data/unit/check-type-aliases.test
+++ b/test-data/unit/check-type-aliases.test
@@ -1318,3 +1318,17 @@ from typing_extensions import TypeAlias
 
 Foo: TypeAlias = ClassVar[int]  # E: ClassVar[...] can't be used inside a type alias
 [builtins fixtures/tuple.pyi]
+
+
+[case testTypeAliasDict]
+D = dict[str, int]
+d = D()
+reveal_type(d)  # N: Revealed type is "builtins.dict[builtins.str, builtins.int]"
+reveal_type(D())  # N: Revealed type is "builtins.dict[builtins.str, builtins.int]"
+reveal_type(D(x=1))  # N: Revealed type is "builtins.dict[builtins.str, builtins.int]"
+reveal_type(D(x="asdf"))  # E: No overload variant of "dict" matches argument type "str" \
+                          # N: Possible overload variants: \
+                          # N:     def __init__(self, **kwargs: int) -> dict[str, int] \
+                          # N:     def __init__(self, arg: Iterable[tuple[str, int]], **kwargs: int) -> dict[str, int] \
+                          # N: Revealed type is "Any"
+[builtins fixtures/dict.pyi]

